ios/FluentUI/Vnext/Button/MSFButton.swift

```diff
@@ -8,91 +8,38 @@ import SwiftUI
-        return self.buttonView.state
-    }
-
-    /// The UIView representing the button.
-    @objc open var view: UIView {
-        return hostingController.view
-    }
-
-    /// Creates a new MSFButton instance.
-    /// - Parameters:
-    ///   - style: The MSFButtonStyle used by the button. Defaults to secondary.
-    ///   - size: The MSFButtonSize value used by the button. Defaults to large.
-    ///   - action: Closure that handles the button tap event.
-    @objc public convenience init(style: MSFButtonStyle = .secondary,
-                                  size: MSFButtonSize = .large,
-                                  action: ((_ sender: MSFButton) -> Void)?) {
-        self.init(style: style,
-                  size: size,
-                  theme: nil,
-                  action: action)
-    }
+    @objc public let state: MSFButtonState
 
     /// Creates a new MSFButton instance.
     /// - Parameters:
     ///   - style: The MSFButtonStyle used by the button. Defaults to secondary.
     ///   - size: The MSFButtonSize value used by the button. Defaults to large.
-    ///   - theme: The FluentUIStyle instance representing the theme to be overriden for this button.
     ///   - action: Closure that handles the button tap event.
     @objc public init(style: MSFButtonStyle = .secondary,
                       size: MSFButtonSize = .large,
-                      theme: FluentUIStyle? = nil,
                       action: ((_ sender: MSFButton) -> Void)?) {
-        super.init()
-        initialize(style: style,
-                   size: size,
-                   action: action,
-                   theme: theme)
-    }
-
-    // MARK: - UIGestureRecognizerDelegate
-
-    public func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer,
-                                  shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer) -> Bool {
-        // Prevents the Large Content Viewer gesture recognizer from cancelling the SwiftUI button gesture recognizers.
-        return true
-    }
-
-    // MARK: - FluentUIWindowProvider
-
-    var window: UIWindow? {
-        return self.view.window
-    }
-
-    // MARK: - Private members
-
-    private func initialize(style: MSFButtonStyle = .secondary,
-                            size: MSFButtonSize = .large,
-                            action: ((_ sender: MSFButton) -> Void)?,
-                            theme: FluentUIStyle? = nil) {
         self.action = action
-        buttonView = FluentButton(style: style,
-                                  size: size) { [weak self] in
+        let buttonView = FluentButton(style: style,
+                                      size: size,
+                                      action: {})
+        state = buttonView.state
+        super.init(AnyView(buttonView))
+
+        // After initialization, set the new action to refer to our own.
+        buttonView.state.action = { [weak self] in
```

Comment:
Not sure I understand why we're setting the action in the state later instead of in line 31.

Reply:
Ah yeah, sadly it's a compiler thing. We create `buttonView` before calling `super.init(_:)`, since that call requires a fully-formed `View`. But the action requires a reference to `self`, which we're not allowed to make until after calling `super.init(_:)`.